test_integration/lib/proof_utils.js

```diff
@@ -21,10 +21,12 @@
 const rlp = require('rlp');
 
 // This is the position of message outbox defined in GatewayBase.sol
-const MESSAGE_OUTBOX_OFFSET = '7';
+// @dev, this is in hex value (in string) without 0x prefix.
+const MESSAGE_OUTBOX_OFFSET = '9';
```

Comment:
Now constants are public, there shouldn't be any constant on JS layer.

Reply:
Let us have this as constant. For testing, it will be better to be a hardcoded value instead of getting from the contract. If it changes in the contract this test should break.

Reply:
With the getter, we want to achieve backward compatibility right? Even if the position will change, it shouldn't impact the flow right?  isn't that the whole point of having explicit getter? 

Integration test in my mind is more like a user journey test, it shouldn't break if the implementation is intended as backward compatible. 
If a test breaks mean you need to act in some way like update some other code, documentation etc. But I believe this is not the case here.